stages/org.osbuild.zip

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/python3
+import os
+import subprocess
+import sys
+
+import osbuild.api
+
+
+def main(inputs, output_dir, options):
+    tree = inputs["tree"]["path"]
+    filename = options["filename"].lstrip("/")
+    # The zip command itself uses 6 as default if no level is specified.
+    # We choose to specifiy 6 as default here to make it explicit.
+    level = options.get("level", 6)
+    paths = options.get("paths", [])
+
+    # Set up the zip command.
+    zip_cmd = [
+        "zip",
+        "-r",
+        f"-{level}",
+        os.path.join(output_dir, filename),
+        "."
+    ]
+
+    if paths:
+        zip_cmd += ["-i"]
+        zip_cmd += paths
+
+    # Unlike the tar command, zip does not allow an option to change working dir.
+    os.chdir(tree)
```

Comment:
Maybe this could just be part of `subprocess.run()` via `cwd=tree` there, i.e.:
```python
diff --git a/stages/org.osbuild.zip b/stages/org.osbuild.zip
index 821f3180..59744ba4 100755
--- a/stages/org.osbuild.zip
+++ b/stages/org.osbuild.zip
@@ -27,13 +27,11 @@ def main(inputs, output_dir, options):
         zip_cmd += ["-i"]
         zip_cmd += paths
 
-    # Unlike the tar command, zip does not allow an option to change working dir.
-    os.chdir(tree)
-
     subprocess.run(
         zip_cmd,
         stdout=subprocess.DEVNULL,
-        check=True
+        check=True,
+        cwd=tree,
     )
 
     return 0
```
This would feel slightly cleaer to me as it only changes the dir inside the subprocess - it really does not matter here but feels slightly more correct :)